model/patch/db_test.go

```diff
@@ -75,7 +75,130 @@ func TestMostRecentByUserAndProject(t *testing.T) {
+	mergeTestPatchPR := Patch{
+		Id:          bson.NewObjectId(),
+		Project:     "evergreen",
+		Description: "Merge Test Patch - PR",
+		Alias:       "fake alias",
+		GithubPatchData: thirdparty.GithubPatch{
+			MergeCommitSHA: "merge_commit_sha_value", // indicates merge_test requester
+		},
+	}
+	assert.NoError(t, mergeTestPatchPR.Insert())
+	patchRequestPatch := Patch{
+		Id:          bson.NewObjectId(),
+		Project:     "evergreen",
+		Description: "Patch Request Patch", // patch_request requester
+	}
+	assert.NoError(t, patchRequestPatch.Insert())
 
+	opts := ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.PatchVersionRequester},
+	}
+	ctx := context.TODO()
+	patches, count, err := ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count)
+	assert.Equal(t, 1, len(patches))
+	assert.Equal(t, "Patch Request Patch", patches[0].Description)
+
+	opts = ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.GithubPRRequester},
+	}
+	patches, count, err = ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count)
+	assert.Equal(t, 1, len(patches))
+	assert.Equal(t, "GH PR Patch", patches[0].Description)
+
+	opts = ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.GithubMergeRequester},
+	}
+	patches, count, err = ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count)
+	assert.Equal(t, 1, len(patches))
+	assert.Equal(t, "GH Merge Patch", patches[0].Description)
+
+	opts = ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.MergeTestRequester},
+	}
+	patches, count, err = ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, count)
+	assert.Equal(t, 2, len(patches))
+	assert.Equal(t, "Merge Test Patch - Alias", patches[0].Description)
+	assert.Equal(t, "Merge Test Patch - PR", patches[1].Description)
+
+	opts = ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.PatchVersionRequester},
+	}
+	patches, count, err = ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count)
+	assert.Equal(t, 1, len(patches))
+	assert.Equal(t, "Patch Request Patch", patches[0].Description)
+
+	opts = ByPatchNameStatusesCommitQueuePaginatedOptions{
+		Project:    utility.ToStringPtr("evergreen"),
+		Requesters: []string{evergreen.PatchVersionRequester, evergreen.GithubMergeRequester},
+	}
+	patches, count, err = ByPatchNameStatusesCommitQueuePaginated(ctx, opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, count)
+	assert.Equal(t, 2, len(patches))
```

Comment:
`require.Equal` or else the next line can panic if `patches` doesn't have 2 items.